pkg/sources/github/github.go

```diff
@@ -575,24 +572,23 @@ func (s *Source) getGistsByUser(ctx context.Context, apiClient *github.Client, u
 	gistURLs := []string{}
 	gistOpts := &github.GistListOptions{}
 	for {
-		gists, resp, err := apiClient.Gists.List(ctx, user, gistOpts)
-		if err == nil {
-			defer resp.Body.Close()
-		}
-		if handled := handleRateLimit(err, resp); handled {
+		gists, res, err := apiClient.Gists.List(ctx, user, gistOpts)
+		res.Body.Close()
+		if handled := handleRateLimit(err, res); handled {
 			continue
 		}
 		if err != nil {
 			log.WithError(err).Warnf("could not list repos for user %s", user)
 			return nil, fmt.Errorf("could not list repos for user %s: %w", user, err)
 		}
+		s.log.Debugf("Listed gists for user %s page %d/%d", user, gistOpts.Page, res.LastPage)
```

Comment:
linter: Pointing out you might have a nil pointer dereference here, might need to check that res != nil first.

Reply:
Oh, I see there is a nil check, i'll move it down.